queries/search-query.ts

```diff
@@ -0,0 +1,167 @@
+    };
+  }
+
+  logger.info(`[Search] ${address} is on XMTP, fetching profiles`);
+  const profiles = await searchProfilesForCurrentAccount(address);
+
+  if (!isEmptyObject(profiles)) {
+    logger.info(`[Search] Found profiles for ${address}, setting profile data`);
+    setProfileRecordSocialsQueryData(profiles);
+    return {
+      message: "",
+      profileSearchResults: profiles,
+    };
+  }
+
+  logger.info(
+    `[Search] No profiles found for ${address}, returning just address`
+  );
+  const justAddress: Record<string, IProfileSocials> = {
+    [address]: { address },
+  };
+  return {
+    message: "address is on xmtp but not on converse yet ;)",
+    profileSearchResults: justAddress,
+  };
+}
+
+/**
+ * Handles searching when searchQuery is a regular string (not a peer)
+ */
+async function handleGeneralSearch(
+  searchQuery: string
+): Promise<ISearchResult> {
+  logger.info(`[Search] Searching profiles for query: ${searchQuery}`);
+  const profiles = await searchProfilesForCurrentAccount(searchQuery);
+
+  if (!isEmptyObject(profiles)) {
+    logger.info(`[Search] Found profiles for query ${searchQuery}`);
+    setProfileRecordSocialsQueryData(profiles);
+    const filteredProfiles = { ...profiles };
+    return {
+      message: "",
+      profileSearchResults: filteredProfiles,
+    };
+  }
+
+  logger.info(`[Search] No profiles found for query ${searchQuery}`);
+  return {
+    message: `They're not here\nInvite them?`,
+    profileSearchResults: {},
+  };
+}
+
+export async function searchUsers({
+  searchQuery,
+}: {
+  searchQuery: string;
+}): Promise<ISearchResult> {
+  logger.info(`[Search] Starting search for query: ${searchQuery}`);
+
+  if (searchQuery.length === 0) {
+    return {
+      message: "",
+      profileSearchResults: {},
+    };
+  }
+
+  if (isSupportedPeer(searchQuery)) {
+    return handlePeerSearch(searchQuery);
+  }
+
+  return handleGeneralSearch(searchQuery);
+}
+
+const SearchResultStaleTime = 1000 * 60 * 5; // 5 minutes
+export function getSearchQueryOptions(searchQuery: string) {
+  return queryOptions({
+    queryKey: userSearchQueryKey(searchQuery),
+    queryFn: () => searchUsers({ searchQuery }),
+    enabled: !!searchQuery,
+    staleTime: SearchResultStaleTime,
+  });
+}
+
+export const useSearchQuery = (args: ISearchArgs) => {
+  const { searchQuery } = args;
+  const {
+    data: { profileSearchResults, message } = {},
+    isLoading: areSearchResultsLoading,
+  } = useQuery(getSearchQueryOptions(searchQuery));
+  const currentAccount = getCurrentAccount()!;
+  const currentAccountAddress = getCleanAddress(currentAccount);
+  const allAddressesToOmit = [
+    ...args.addressesToOmit,
+    currentAccountAddress,
+  ].map((address) => address.toLowerCase());
+
+  for (const address of Object.keys(profileSearchResults || {})) {
+    if (allAddressesToOmit.includes(address)) {
+      if (profileSearchResults) {
+        delete profileSearchResults[address];
+      }
+    }
+  }
```

Comment:
we can use the `select` in `useQuery` to do the filtering

Reply:
I don’t want to use that because it would prevent those users from entering the cache and this cache is an installation wide cache used by all users so doing it manually is correct behavior I believe

Reply:
ohhhh nvm i left that comment tired - you right